src/lib/tools/fetchShowsNearLocation.ts

```diff
@@ -0,0 +1,91 @@
+import { tool } from "ai"
+import { extractNodes, metaphysics } from "../metaphysics"
+import { z } from "zod"
+
+const DEFAULT_NUMBER_OF_SHOWS = 100
+const MAX_NUMBER_OF_SHOWS = 200
+
+export const fetchShowsNearLocation = tool({
+  description: `
+    Fetch a list of exhibitions or shows near a given lat/lng location
+  `,
+
+  parameters: z.object({
+    longitude: z
+      .number()
+      .min(-180)
+      .max(180)
+      .default(DEFAULT_NUMBER_OF_SHOWS)
+      .describe(`The longitude of the location to search near`),
+    latitude: z
+      .number()
+      .min(-90)
+      .max(90)
+      .default(DEFAULT_NUMBER_OF_SHOWS)
+      .describe(`The latitude of the location to search near`),
+    size: z
+      .number()
+      .max(MAX_NUMBER_OF_SHOWS)
+      .default(DEFAULT_NUMBER_OF_SHOWS)
+      .describe(`The maximum number of shows to return`),
```

Comment:
Because of the way that the AI SDK incorporates the Zod schema, it serves nicely as inline documentation, dev-time intellisense _and_ runtime safety. So much winning you will get tired of winning.

An example of user input that causes the model to call a function with disallowed params:

<pre>
InvalidToolArgumentsError [AI_InvalidToolArgumentsError]: Invalid arguments for tool fetchShowsNearLocation: Type validation failed: Value: {"latitude":40.8503876,"longitude":-73.93699079999999,"size":25}.
Error message: [
  {
    <b>"code": "too_big",</b>
    <b>"maximum": 20,</b>
    "type": "number",
    "inclusive": true,
    "exact": false,
    <b>"message": "Number must be less than or equal to 20",</b>
    "path": [
      "size"
    ]
  }
]
</pre>